dhcpv4/client.go

```diff
@@ -165,14 +207,110 @@ func (c *Client) Exchange(ifname string, discover *DHCPv4, modifiers ...Modifier
 	conversation = append(conversation, request)
 
 	// Ack
-	ack, err := BroadcastSendReceive(sfd, rfd, request, c.ReadTimeout, c.WriteTimeout, MessageTypeAck)
+	ack, err := c.SendReceive(sfd, rfd, request, MessageTypeAck)
 	if err != nil {
 		return conversation, err
 	}
 	conversation = append(conversation, ack)
 	return conversation, nil
 }
 
+// SendReceive sends a packet (with some write timeout) and waits for a
+// response up to some read timeout value. If the message type is not
+// MessageTypeNone, it will wait for a specific message type
+func (c *Client) SendReceive(sendFd, recvFd int, packet *DHCPv4, messageType MessageType) (*DHCPv4, error) {
+	var laddr net.UDPAddr
+	if c.LocalAddr == nil {
+		laddr = net.UDPAddr{IP: net.IPv4bcast, Port: ServerPort}
+	} else {
+		if addr, ok := c.LocalAddr.(*net.UDPAddr); ok {
+			laddr = *addr
+		} else {
+			return nil, fmt.Errorf("Invalid local address: not a net.UDPAddr: %v", c.LocalAddr)
```

Comment:
DHCPv4 also supports TCP, but we don't (yet?), a more accurate error message would be 
```
Invalid local address, only net.UDPAddr is supported, got %v instead", reflect.TypeOf(c.LocalAddr))
```